### ISSUE TITLE
immer does not work in react18

### ISSUE THREAD

**Issue by cutefcc** [pmndrs/jotai#1164]
```import { useAtom } from "jotai";
import { atomWithImmer, useImmerAtom } from "jotai/immer";

const store = atomWithImmer({ str: "init str" });

function TestJotai() {
  const [obj, setObj] = useImmerAtom(store);
  console.log("TestJotai -- render", obj.str);

  const changeFn = () => {
    // write like this, why immer not working
    setObj((draft) => {
      draft.str = "init str new";
    });
  };
  return (
    <div>
      <h2>TestJotai</h2>
      <p>str: {obj.str}</p>
      <button onClick={changeFn}>changeState</button>
    </div>
  );
}

export default TestJotai;
```
When I keep clicking the button, the component will be Re-rendered repeatedly，I think it's unnecessary

https://user-images.githubusercontent.com/63641648/168634246-210868bf-795b-49c0-9b56-92a966913169.mov

**Comment by dai-shi** [comment]
This is probably a new behavior introduced in React 18. It shouldn't affect the performance much.

Can you try if #1159 fixes your issue?
https://ci.codesandbox.io/status/pmndrs/jotai/pr/1159 👈 Find "Local Install Instructions"

btw, using both `atomWithImmer` and `useImmerAtom` seems redundant.

**Comment by TwistedMinda** [comment]
I can't reproduce the issue.

With all the latest version the render happens only if value changed as expected.

```tsx
{
  "immer": "^9.0.14",
  "jotai": "^1.6.6",
  "react": "^18.1.0",
}
```

**Comment by dai-shi** [comment]
@TwistedMinda Thanks for trying to reproduce it!

closing for now as #1159 is shipped.
we can reopen with a reproduction.

**Comment by cutefcc** [comment]
@dai-shi I try https://github.com/pmndrs/jotai/pull/1159,  I find "Local Install Instructions" reinstalled jotai like this:  npm i https://pkg.csb.dev/pmndrs/jotai/commit/15d58d06/jotai

still doesn't solve my problem

**Comment by dai-shi** [comment]
Please wrap your `console.log` in `useEffect` and see if the result is as expected.

If that doesn't work, please create a minimal reproduction with codesandbox.

**Comment by cutefcc** [comment]
@dai-shi

**Comment by cutefcc** [comment]
@dai-shi  After I deleted node__modules and reinstalled npm i https://pkg.csb.dev/pmndrs/jotai/commit/15d58d06/jotai, it took effect, thank you very much,


Finally I test:
- "jotai": "1.6.6" immer will not work
- "jotai": "https://pkg.csb.dev/pmndrs/jotai/commit/c6800061/jotai", yes, it meets my expectations

When will jotai upgrade to solve this problem? instead of installing via https://pkg.csb.dev/pmndrs/jotai/commit/c6800061/jotai, which looks weird

**Comment by cutefcc** [comment]
@dai-shi  Just now, I found that "jotai": "1.6.7", no longer has this problem. Was it fixed in 1.6.7?

**Comment by dai-shi** [comment]
Yes, #1159 is included in https://github.com/pmndrs/jotai/releases/tag/v1.6.7.

But, this is not technically a big problem. v1.6.6 works fine too. #1159 does a **little** optimization mainly to avoid people complaining. Final performance may or may not change very much.
